### DISCUSSION TITLE
PPR and CDN Caching

### DISCUSSION THREAD

**Discussion by khuezy** [vercel/next.js#58350] [Help]
### Summary

Hello, Vercel's demo of PPR: https://vercel.com/templates/next.js/partial-prerendering-nextjs
The response `cache-control` header for the home page is: `public, max-age=0, must-revalidate`

But when I check out the repo and `run build/start`, the `cache-control` is: `s-maxage=31536000, stale-while-revalidate`

Is PPR a Vercel only feature? When I deploy my app in a container with NGINX caching the page, it caches dynamic data (eg session data)... Am I missing something?

TIA

**Comment by khuezy** [comment]
This might fix the cache-control issue I was seeing: https://github.com/vercel/next.js/pull/59447

**Reply by conico974** [reply]
Just wanted to add something here.
PPR are not served from the CDN at all in standalone. The only place where PPR is supported the way they showed on the demo is still on Vercel (And that will likely stay like that unless they change PPR to client side rendering instead of SSR with streaming).
With the fix you'll still hit the next server for every SSR request ( NextServer will use the prerendered html )

To replicate what they do in Vercel, you'll have to serve the prerendered html from the edge (something like cloudflare workers) and transform this html to insert the postponed RSC (which is undocumented) while streaming the response. It might also be necessary to use the undocumented minimal mode of NextServer

**Reply by apatel369** [reply]
@conico974 Next.js doc says that static shell is served from the nearest edge region. So it uses CDN for pre-rendered HTML and NextServer only for serving dynamic parts?

https://vercel.com/blog/partial-prerendering-with-next-js-creating-a-new-default-rendering-model
"When a user visits a page, the fast static shell is served from the end-user’s nearest [Edge Region](https://vercel.com/docs/edge-network/overview)"

**Reply by conico974** [reply]
@apatel369 That is not how this works in standalone or with `next start`. 
This works like that only if you host on Vercel, if you host in a different way, your request has to hit the server.
The server fetch the static shell from the filesystem ( Or from somewher else if you use an external cache handler), then it starts streaming, but everything happens on the server.

**Reply by apatel369** [reply]
> @apatel369 That is not how this works in standalone or with `next start`. 
> 
> This works like that only if you host on Vercel, if you host in a different way, your request has to hit the server.
> 
> The server fetch the static shell from the filesystem ( Or from somewher else if you use an external cache handler), then it starts streaming, but everything happens on the server.

That makes sense. Thanks for explaining.
